---
title: "ACC 100bp intervals differential methylation genomewide analysis"
author: "Alexander Favorov"
output: 
 html_document:
  toc: true 

---

`r knitr::opts_chunk$set(echo=F,message=F,cache=F,warning=F)`
<!-- cache is off, we save all the neccessary things in Rda -->


### `r format(Sys.Date(), "%d %b %Y")` 

Here, we run an unsuperviser 100bp-probe genomwide differential methylation test.

### The list of all differentially methylated (p<=0.05) C-noodles with Bonferroni correction

Here is <a href=./DM.C.noodles.bonf.strict.bed> strict BED file with the DM interval list</a> and the <a href=./DM.C.noodles.bonf.bed> bed file with p-values </a>. 

The genes that TSS overlap with these intervals are summarised <a href=./DM.C.noodles.bonf.adjacent.genes.tsv>here</a>.
Here are the lists of genes that TSS overlap with DM with expanding by <a href=./DM.C.noodles.bonf.adjacent.genes.100000.tsv>100000 bp</a> flanks and by <a href=./DM.C.noodles.bonf.adjacent.genes.1000000.tsv>1000000 bp</a> flanks.

### The list of all differentially methylated (FDR<=0.1) C-noodles with FDR correction

Here is <a href=./DM.C.noodles.fdr.strict.bed> strict BED file with the DM interval list</a> and the <a href=./DM.C.noodles.fdr.bed> bed file with p-values </a>. 

The genes that TSS overlap with these intervals are summarised <a href=./DM.C.noodles.fdr.adjacent.genes.tsv>here</a>.
Here are the lists of genes that TSS overlap with DM with expanding by <a href=./DM.C.noodles.fdr.adjacent.genes.100000.tsv>100000 bp</a> flanks and by <a href=./DM.C.noodles.fdr.adjacent.genes.1000000.tsv>1000000 bp</a> flanks.


